app/models/message_summarizer/message_summary.rb

```diff
@@ -0,0 +1,33 @@
+class MessageSummarizer::MessageSummary
+  attr_reader :controller
+
+  def initialize(controller)
+    @controller = controller
+  end
+
+  def any?
+    count > 0
+  end
+
+  def count
+    @count ||= allowed? ? get_count : 0
+  end
+
+  def link
+    controller.view_context.link_to(label, path)
+  end
```

Comment:
It might be worth adding the methods with `NotImplementedError`s for the methods we expect to be overridden.